### DISCUSSION TITLE
Can I use the methods which I have in ServerToDoItemService and call them from my ToDoItemController?

### DISCUSSION THREAD

**Discussion by vnetonline** [oqtane/oqtane.framework#4934] [Q&A]
@sbwalker I am working on a simple ToDoItem course for [LearnOqtane.com](https://learnoqtane.com/) to demonstrate how to build a real life module using Oqtane Framework.

My question is can I use the methods which I have in my ServerToDoItemService in my ToDoItemController? rather then reimplementing the functionality again in controller?

Course to be released soon on [LearnOqtane.com](https://learnoqtane.com/) not a member? [Join Now](https://learnoqtane.com/join-now) for FREE

**Comment by sbwalker** [comment]
@vnetonline have you scaffolded a default module from within Oqtane?

- login as host
- navigate to Admin Dashboard / Module Management
- click Create Module
- enter required information and click Create Module

If you look at the code that is created you will see that the Controller is a very thin class - it simply declares the endpoints (ie. methods) and  some security concerns - but the actual logic for interacting with the database is implemented in the Service class in the Server project.

**Reply by vnetonline** [reply]
@sbwalker I have done exactly that but the controller injects the repository which is what the server service implementation is also doing

That's why I asked the question because it is replicating the logic that server service uses so why not inject the server service and utilise it in the controller??

**Reply by sbwalker** [reply]
Yes the Controller should call the Server Service to avoid duplication - this should be changed in the default Module template to match the approach used by SiteController/SiteServerService in the framework

**Reply by sbwalker** [reply]
@vnetonline #4952 resolves the issue you identified in the default module template
